lib/charty/backends.rb

```diff
@@ -0,0 +1,58 @@
+module Charty
+  class BackendError < RuntimeError; end
+  class BackendNotFoundError < BackendError; end
+  class BackendLoadError < BackendError; end
+
+  module Backends
+    @backends = {}
+
+    def self.names
+      @backends.keys
+    end
+
+    def self.register(name, backend_class)
+      case name
+      when Symbol
+        name = name.to_s
+      else
+        name = name.to_str
+      end
+      @backends[name] = {
```

Comment:
Should we check the redefinition of the same backend name?

Reply:
We can do it but it's not important. Because backend developers don't do so.
If this is used by normal Charty users, it's better that this is implemented.

Reply:
OK, I want to postpone the decision.